doc/roadmap-2024.md

```diff
@@ -0,0 +1,75 @@
+# TiKV Roadmap (2024~2025)
+
+- Last Updated: 2024-04-23
+
+This is a roadmap of TiKV project from engineering and architecture aspects. It expresses the intension of how we will improve this project to make TiKV fit more users' use cases in next 1 to 2 years. The roadmap includes several domains like latency and performance improvement, stability, observability, etc. Notice: all content list in the roadmap is our intension, we will try our best efforts to make it happen, but it doesn't mean we have a commitment for it, please don't rely on features/capabilities currently TiKV doesn't have for your projects which have strict timeline requirement, even these features listed in the roadmap.
+
+## QoS: Stable and Predictable Latency
+
+This domain aims to enhance TiKV to provide predictable single-digit millisecond latency under different scenarios like too many MVCC versions, disk IO temporarily jitter, high read QPS in a small data range, etc. 
+
+### In-Memory Engine
```

Comment:
would it make sense to add a comment about other uses for in-memory usage?

Reply:
We can add it later since we don't know what other scenario can benefit from in-memory engine now.